go/chat/storage/storage.go

```diff
@@ -18,6 +18,13 @@ import (
 // ***
 const cryptoVersion = 1
 
+type resultCollector interface {
```

Comment:
`simpleResultCollector` and `typedResultCollector` look pretty similar. I'm not sure the `resultCollector` abstraction is worth it. Do you anticipate any more implementations? Maybe a `resultCollector.useTypMap bool` would be enough. Your call though.